modules/net-vpc-swp/main.tf

```diff
@@ -0,0 +1,119 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  create_url_lists = { for k, v in var.policy_rules.url_lists : v.url_list => v if v.values != null }
+}
+
+resource "google_network_security_gateway_security_policy" "policy" {
+  provider              = google-beta
+  project               = var.project_id
+  name                  = var.name
+  location              = var.region
+  description           = "Managed by Terraform."
+  tls_inspection_policy = var.tls_inspection_config != null ? google_network_security_tls_inspection_policy.tls-policy.0.id : null
+}
+
+resource "google_network_security_tls_inspection_policy" "tls-policy" {
+  count                 = var.tls_inspection_config != null ? 1 : 0
+  provider              = google-beta
+  project               = var.project_id
+  name                  = var.name
+  location              = var.region
+  ca_pool               = var.tls_inspection_config.ca_pool
+  exclude_public_ca_set = var.tls_inspection_config.exclude_public_ca_set
+}
+
+resource "google_network_security_gateway_security_policy_rule" "secure_tag_rules" {
+  for_each                = var.policy_rules.secure_tags
+  provider                = google-beta
+  project                 = var.project_id
+  name                    = each.key
+  location                = var.region
+  gateway_security_policy = google_network_security_gateway_security_policy.policy.name
+  enabled                 = each.value.enabled
+  priority                = each.value.priority
+  session_matcher         = <<-EOT
+  source.matchTag('${each.value.tag}')%{if each.value.session_matcher != null} && (${each.value.session_matcher})%{endif~}
+  EOT
+  application_matcher     = each.value.application_matcher
+  tls_inspection_enabled  = each.value.tls_inspection_enabled
+  basic_profile           = each.value.action
+}
+
+resource "google_network_security_url_lists" "url_lists" {
+  for_each    = local.create_url_lists
+  provider    = google-beta
+  project     = var.project_id
+  name        = each.key
+  location    = var.region
+  description = "Managed by Terraform."
+  values      = each.value.values
+}
+
+resource "google_network_security_gateway_security_policy_rule" "url_list_rules" {
+  for_each                = var.policy_rules.url_lists
+  provider                = google-beta
+  project                 = var.project_id
+  name                    = each.key
+  location                = var.region
+  gateway_security_policy = google_network_security_gateway_security_policy.policy.name
+  enabled                 = each.value.enabled
+  priority                = each.value.priority
+  session_matcher         = <<EOT
```

Comment:
Can you indent this heredoc like you did with the other one (`<<-EOT`)

Reply:
I tried but it feels like a Terraform bug that it doesn't strip all the whitespace correctly with the indented version and it results in whitespaces that mess with the tests.

Reply:
Looks like this:
```
E             assert "inUrlList(ho...ccount.com'))" == "inUrlList(ho...ccount.com'))"
E               - inUrlList(host(), 'projects/my-project/locations/europe-west4/urlLists/my-url-list') && (source.matchServiceAccount('my-sa@my-project.iam.gserviceaccount.com'))
E               + inUrlList(host(), '    projects/my-project/locations/europe-west4/urlLists/my-url-list
E               + ')&& (source.matchServiceAccount('my-sa@my-project.iam.gserviceaccount.com'))
```